handler/src/test/java/io/netty/handler/ssl/SslHandlerTest.java

```diff
@@ -560,4 +564,127 @@ public void operationComplete(Future<Channel> future) {
             ReferenceCountUtil.release(sslClientCtx);
         }
     }
+
+    @Test(timeout = 30000)
+    public void testCompositeBufSizeEstimationGuaranteesSynchronousWrite() throws CertificateException, SSLException {
```

Comment:
also would be nice if we had this as test that not depends on the `SslHandler` but only uses the SSLEngine itself